CMakeLists.txt

```diff
@@ -377,9 +381,10 @@ else(ENABLE_TCP)
 endif(ENABLE_TCP)
 
 # creates config header file in build directory
-configure_file(${CMAKE_CURRENT_LIST_DIR}/include/coap2/coap.h.in ${CMAKE_CURRENT_BINARY_DIR}/include/coap2/coap.h)
+configure_file(${CMAKE_CURRENT_LIST_DIR}/include/coap2/coap.h.in
+               ${CMAKE_CURRENT_BINARY_DIR}/include/coap2/coap.h)
 
-configure_file(${CMAKE_CURRENT_LIST_DIR}/include/coap2/coap_config.h.in
+configure_file(${CMAKE_CURRENT_LIST_DIR}/include/coap2/cmake_coap_config.h.in
```

Comment:
I still do not see why this file should be in `include/coap2` where it does not belong. `${CMAKE_CURRENT_LIST_DIR}/cmake_coap_config.h.in` works like a charm.

Reply:
Currently the only git controlled file that is in the top directory is `libcoap-2.pc.in`.  We have git controlled ` include/coap2/coap.h.in` so  I guess `cmake_coap_config.h.in` could logically be in either place.  I think that I would prefer it to be in `include/coap2` as it ends up as being the included file `coap_config.h`, but happy to make the suggested change if you want.

Reply:
In the top directory, we also have the other special `coap_config.h.*` files. `coap_config.h` is being created in the top-level build directory which is deliberately separated from `include/coap2` as `coap_config.h` does not get installed by `make install`.
`coap.h.in` is different because it is used to create `coap.h` which belongs into `include/coap2`. Therefore, it makes sense for `coap.h.in` to have the template in the same place as the created file.

I really prefer not to have `cmake_config.h.in` not to have in `include/coap2`. The question is if we might want to have a dedicated sub-directory for the different configs? (If yes, I propose putting the cmake config in the top directory for now to do the merge of this PR.)

Reply:
`cmake_config.h.in` is now in the top directory and code pushed.